proxy/src/bin/proxy.rs

```diff
@@ -382,27 +381,29 @@ async fn main() -> anyhow::Result<()> {
 
     let cancellation_token = CancellationToken::new();
 
-    let cancel_map = CancelMap::default();
-
     let redis_rps_limit = Vec::leak(args.redis_rps_limit.clone());
     RateBucketInfo::validate(redis_rps_limit)?;
 
-    let redis_publisher = match &regional_redis_client {
-        Some(redis_publisher) => Some(Arc::new(Mutex::new(RedisPublisherClient::new(
+    let max_redis_rps = redis_rps_limit
+        .iter()
+        .map(|x| x.rps() as usize)
+        .max()
+        .unwrap_or(100_000); // this is the global proxy limit
```

Comment:
This feels quite high for a channel size. It would only be 11MB at max, but I think it's still way too high.

I think it would be better to set this much lower (eg 64). The logic is that redis should ideally be able to keep up, and if it can't we won't find out until much later.

Reply:
I choose the channel size so high because of the peak load. Agree, that 100k is really high (and 200k for channel size). I checked the benchmarks and the typical value is about 1-2 ms per redis op (we have at least 2 redis ops for each request). So, the normal processing should be <1k. I'll change the channel size to 1k, but the redis limit itself I think should be close to the max proxy limit, which I think should be much higher than 1k, like 10 or 100k. Wdyt?

Reply:
I still think setting it even lower is better. Sourcing an additional opinion from @cloneable

Reply:
Sorry, need a bit more time. Quite a lot of code.

Depends on what we're protecting against. We could say that we want to buffer up to one minute of cancellation data. Or 30 seconds.

But the primary purpose of the channel is to not delay entering proxy passthrough and send the cancellation info in the background?

Reply:
@cloneable yes, to make it predictable

Reply:
Btw, `handle_messages` will rapidly consume whatever is in the channel and drop RedisOps when there's a Redis error, rate-limit exceeded, or Redis disconnected and reconnect failed, right? It's up to the original caller sending the RedisOp to re-enqueue the request if retry is wanted?

Reply:
Yes, that's correct

Reply:
@cloneable so, what's your opinion on the channel limits?

Reply:
Agree, didn't notice that timeout is 10sec, I thought I set 10msec :) Thanks, fixed. As a channel metric I've added cancellation channel size as a counter pair.